compiler/rustc_traits/src/evaluate_obligation.rs

```diff
@@ -15,18 +17,32 @@ fn evaluate_obligation<'tcx>(
     tcx: TyCtxt<'tcx>,
     canonical_goal: CanonicalPredicateGoal<'tcx>,
 ) -> Result<EvaluationResult, OverflowError> {
-    debug!("evaluate_obligation(canonical_goal={:#?})", canonical_goal);
-    // HACK This bubble is required for this tests to pass:
-    // impl-trait/issue99642.rs
-    let (ref infcx, goal, _canonical_inference_vars) = tcx
-        .infer_ctxt()
-        .with_opaque_type_inference(DefiningAnchor::Bubble)
-        .build_with_canonical(DUMMY_SP, &canonical_goal);
-    debug!("evaluate_obligation: goal={:#?}", goal);
-    let ParamEnvAnd { param_env, value: predicate } = goal;
+    if tcx.sess.opts.unstable_opts.trait_solver != TraitSolver::Next {
+        debug!("evaluate_obligation(canonical_goal={:#?})", canonical_goal);
+        // HACK This bubble is required for this tests to pass:
+        // impl-trait/issue99642.rs
+        let (ref infcx, goal, _canonical_inference_vars) = tcx
+            .infer_ctxt()
+            .with_opaque_type_inference(DefiningAnchor::Bubble)
+            .build_with_canonical(DUMMY_SP, &canonical_goal);
+        debug!("evaluate_obligation: goal={:#?}", goal);
+        let ParamEnvAnd { param_env, value: predicate } = goal;
 
-    let mut selcx = SelectionContext::with_query_mode(&infcx, TraitQueryMode::Canonical);
-    let obligation = Obligation::new(tcx, ObligationCause::dummy(), param_env, predicate);
+        let mut selcx = SelectionContext::with_query_mode(&infcx, TraitQueryMode::Canonical);
+        let obligation = Obligation::new(tcx, ObligationCause::dummy(), param_env, predicate);
 
-    selcx.evaluate_root_obligation(&obligation)
+        selcx.evaluate_root_obligation(&obligation)
+    } else {
+        let Ok(response) = TyCtxtSolveExt::evaluate_goal(
+            tcx,
+            canonical_goal.unchecked_map(|pred| pred.into()),
+        ) else {
+            return Ok(EvaluationResult::EvaluatedToErr);
+        };
+        match response.value.certainty {
+            Certainty::Yes => Ok(EvaluationResult::EvaluatedToOk),
```

Comment:
can you change that to `EvaluatedToOkModuloRegions` (with a fixme to look at the external region contraints once we've implemented them)